### PR TITLE
Provide a better exception when symbols are not found or not matching & allow skipping those exceptions

### PR DESCRIPTION
**Pull request by jbevain**

### CODE REVIEW

**Comment by jbevain** [comment]
This addresses #480.

@spouliot, could your review? :)

File: symbols/mdb/Mono.Cecil.Mdb/MdbReader.cs [hunk 1]
```diff
@@ -33,15 +33,7 @@ public ISymbolReader GetSymbolReader (ModuleDefinition module, Stream symbolStre
 			Mixin.CheckModule (module);
 			Mixin.CheckStream (symbolStream);
 
-			var file = MonoSymbolFile.ReadSymbolFile (symbolStream);
```
Comment (jbevain) [inline symbols/mdb/Mono.Cecil.Mdb/MdbReader.cs]: This is done through ProcessHeader

File: Mono.Cecil.Cil/Symbols.cs [hunk 2]
```diff
@@ -761,6 +761,44 @@ public interface ISymbolReaderProvider {
 		ISymbolReader GetSymbolReader (ModuleDefinition module, Stream symbolStream);
 	}
 
+#if !NET_CORE
+	[Serializable]
+#endif
+	public sealed class SymbolNotFoundException : FileNotFoundException {
```
Comment (spouliot) [inline Mono.Cecil.Cil/Symbols.cs]: minor: there's an inconsistency between `Symbol` and `Symbols` in the API
e.g. `GetSymbolReader` and `ReadSymbols`
I think it's more the later (plural) but I don't care very much - still you should pick an official *winner* before adding more :)

File: Mono.Cecil.Cil/Symbols.cs [hunk 3]
```diff
@@ -761,6 +761,44 @@ public interface ISymbolReaderProvider {
...
+			: base (info, context)
+		{
+		}
+#endif
+	}
+
+#if !NET_CORE
+	[Serializable]
+#endif
+	public sealed class SymbolNotMatchingException : InvalidOperationException {
```
Comment (spouliot) [inline Mono.Cecil.Cil/Symbols.cs]: same: `Symbol[s]`

**Pull request merged**